server/app/com/xsn/explorer/databaseMigrations/BlockRewardsMigration.scala

```diff
@@ -0,0 +1,126 @@
+  private val logger = LoggerFactory.getLogger(this.getClass)
+
+  def run() = {
+    val targetBlock = blockDataHandler.getLatestBlock().toFutureOr
+
+    targetBlock.map { targetBlock =>
+      logger.info(s"Migrating Block Rewards from block 0 to block ${targetBlock.height}")
+
+      val startingState = Future.successful(Good(())).toFutureOr
+      val finalState = (0 to targetBlock.height.int).foldLeft(startingState) {
+        case (state, height) =>
+          for {
+            _ <- state
+            block <- blockDataHandler.getBy(Height(height)).toFutureOr
+            rpcBlock <- xsnService.getFullBlock(block.hash).toFutureOr
+            _ <- block.extractionMethod match {
+              case BlockExtractionMethod.TrustlessProofOfStake =>
+                recalculateExtractionMethod(block, rpcBlock).toFutureOr
+              case _ => Future.successful(Good(())).toFutureOr
+            }
+            _ <- db.getReward(block.hash).toFutureOr.map {
+              case None => storeBlockReward(rpcBlock)
+              case _ => Future.successful(())
+            }
+            _ = Future.successful(Good(logProgress(height, targetBlock.height.int))).toFutureOr
+          } yield ()
+      }
+
+      finalState.toFuture.onComplete {
+        case Success(Good(_)) => logger.info("Block rewards successfully migrated")
+        case Success(Bad(error)) => logger.info(s"Block reward migration failed due to ${error.toString}")
+        case Failure(error) => logger.info(s"Block reward migration failed due to ${error.toString}")
+      }
+    }
+  }
+
+  private def logProgress(migratedHeight: Int, targetHeight: Int) = {
+    val percentage: Int = 100 * migratedHeight / targetHeight
+    val previousPercentage: Int = 100 * (migratedHeight - 1) / targetHeight
+
+    if (percentage != 0 && percentage != previousPercentage) {
+      logger.info(s"migrated block at height ${migratedHeight}, ${percentage}% done")
+    }
+  }
+
+  private def recalculateExtractionMethod(block: Block, rpcBlock: rpc.Block[_]) = {
+    val result = for {
+      extractionMethod <- blockService.extractionMethod(rpcBlock).toFutureOr
+      _ <- db.updateBlock(block.copy(extractionMethod = extractionMethod)).toFutureOr
+    } yield Good(())
+
+    result.future
+  }
+
+  private def storeBlockReward(rpcBlock: rpc.Block[_]) = {
+    val result = for {
+      extractionMethod <- blockService.extractionMethod(rpcBlock).toFutureOr
+      reward <- blockService.getBlockRewards(rpcBlock, extractionMethod).toFutureOr
+      _ <- db.insertBlockRewards(rpcBlock.hash, reward).toFutureOr
+    } yield Good(())
+
+    result.future
+  }
+}
+
+object BlockRewardsMigration {
+  private class DatabaseOperations @Inject()(
+      override val database: Database,
+      blockPostgresDAO: BlockPostgresDAO,
+      blockRewardPostgresDAO: BlockRewardPostgresDAO
+  )(
+      implicit dbEC: DatabaseExecutionContext
+  ) extends AnormPostgresDataHandler {
+
+    def updateBlock(block: Block) = {
```

Comment:
There is no need to update the whole block, just write the small sql to update the extraction method only.